### PR TITLE
perf(consensus): Undo revert #3211. (Remove reactor Consensus RLocks)

### PR DESCRIPTION
**Pull request by ValarDragon**
Pretty simple bug fix for the e2e failure on #3211. There was a race condition at iniitialization for initial height, because we didn't initialize it early on enough.

The error in the E2E logs was:
```
validator03    | E[2024-06-21|21:13:20.744] Stopping peer for error                      module=p2p peer="Peer{MConn{10.186.73.2:34810} 4fe295e4cfad69f1247ad85975c6fd87757195db in}" err="invalid field LastCommitRound can only be negative for initial height 0"
validator03    | I[2024-06-21|21:13:20.744] service stop                                 module=p2p peer=4fe295e4cfad69f1247ad85975c6fd87757195db@10.186.73.2:34810 msg="Stopping Peer service" impl="Peer{MConn{10.186.73.2:34810} 4fe295e4cfad69f1247ad85975c6fd87757195db in}"
```
hinting at initial height not being set rpoperly.

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments
- [ ] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

File: internal/consensus/reactor.go [hunk 1]
```diff
@@ -57,10 +58,11 @@ type ReactorOption func(*Reactor)
...
 func NewReactor(consensusState *State, waitSync bool, options ...ReactorOption) *Reactor {
 	conR := &Reactor{
-		conS:     consensusState,
-		waitSync: atomic.Bool{},
-		rs:       consensusState.GetRoundState(),
-		Metrics:  NopMetrics(),
+		conS:          consensusState,
+		waitSync:      atomic.Bool{},
+		rs:            consensusState.GetRoundState(),
+		initialHeight: consensusState.state.InitialHeight,
```
Comment (ValarDragon) [inline internal/consensus/reactor.go:64]: This line is what was missing in the original PR, and is what should fix the bug

**Comment by ValarDragon** [comment]
e2e-manual run: https://github.com/cometbft/cometbft/actions/runs/9671388479

**Review by melekes — APPROVED** [review]
Thanks @ValarDragon ❤️

**Comment by melekes** [comment]
_Blocked until we release an RC._

File: internal/consensus/reactor.go (hunk 1 above)
Comment (cason) [reply internal/consensus/reactor.go:64]: From the family, always initialize any variable you add, ehehe

**Review by cason — APPROVED** [review]
The diffs match, the only new part is line 64, where we initialize `initialHeight`.

I really hope we don't need to revert the reversion of the reversion of it.

**Comment by melekes** [comment]
@mergify backport v1.x

**Pull request merged**